### ISSUE TITLE
Stage 2 Coil optimisation example

### ISSUE THREAD

**Issue by florianwechsung** [hiddenSymmetries/simsopt#47]
In coil optimisation we often have the case that we have a small number of coils (say 3) that is rotated and flipped to obtain a full stellarator. Currently we express this in the following form:

```
base_coils = get_coils(...) # these are 3 coils that we optimise for
base_currents = [...] # the three currents
stellarator = CoilCollection(base_coils, base_currents, nfp=3, stellsym=True)

full_coils = stellarator.coils
full_currents = stellarator.currents
```

the `CoilCollection` object takes care of the rotation and flipping of coils via `RotatedCurve`.
The `dof`/`optimizable` class should be able to deal with the following problem:

We have an objective that takes a list of coils and currents
```
J = SomeObjective(full_coils, full_currents)
```
and then we want to do the optimisation over `base_coils` and `base_currents`.

**Comment by florianwechsung** [comment]
In addition, it would be good if one could optimise for things like 

`SomeObjective(full_coils, full_currents) + AnotherObjective(base_coils[0]) + AnotherObjective(base_coils[1]) + AnotherObjective(base_coils[2])`
`AnotherObjective` would be something like curvature or coil length.

**Comment by florianwechsung** [comment]
There is a working example of this in `driver.py` of https://github.com/florianwechsung/SIMSOPT-Flux-Minimization 
Requires the `fw/vjp` branch of SIMSOPT.

**Comment by florianwechsung** [comment]
Closed by #130
